runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java

```diff
@@ -134,57 +135,68 @@ public boolean TXActive() {
      * @throws TransactionAbortedException If the transaction could not be executed successfully.
      */
     @SuppressWarnings({"checkstyle:methodname", "checkstyle:abbreviation"})
-    public long TXEnd()
-            throws TransactionAbortedException {
+    public long TXEnd() throws TransactionAbortedException {
         AbstractTransactionalContext context = TransactionalContext.getCurrentContext();
         if (context == null) {
             log.warn("Attempted to end a transaction, but no transaction active!");
             return AbstractTransactionalContext.UNCOMMITTED_ADDRESS;
-        } else {
-            long totalTime = System.currentTimeMillis() - context.getStartTime();
-            log.trace("TXEnd[{}] time={} ms", context, totalTime);
+        }
+
+        // Continue with ending the transaction if the context is not null
+        long totalTime = System.currentTimeMillis() - context.getStartTime();
+        log.trace("TXEnd[{}] time={} ms", context, totalTime);
+
+        // If exist, stop the timer for timing the beginning of transaction to start of commit.
+        if (context.getTxOpDurationContext() != null) {
+            context.getTxOpDurationContext().stop();
+        }
+
+        // Create a timer to measure the transaction commit duration
+        Timer txCommitDurationTimer = context
+                .getMetrics()
+                .timer(CorfuComponent.OBJECT.toString() + "txn-commit-duration");
```

Comment:
Can you please create a constant for this timer name somewhere and reuse in your code